python/pyspark/testing/sqlutils.py

```diff
@@ -0,0 +1,268 @@
+#
```

Comment:
Maybe rename this file to `sql_testing_utils.py`? And is it possible to just place this in the `sql/tests` directory? Having the `testing` folder seems a little ambiguous to me

Reply:
Yea, similar thought. One thing is though testing util itself has some dependencies to each others (for instance, `from pyspark.tests import ReusedPySparkTestCase`). So, I thought it's okay to make it separate.

Another benefit from doing so is we can simply refer the example (NumPy) as is. As you already know, NumPy has been there more then 15 years and should be mature enough. It should be okay to trust their approach.